lib/Backends/OpenCL/OpenCL.cpp

```diff
@@ -1578,6 +1578,19 @@ void OpenCLFunction::translateTraceEvents(ExecutionContext *context) const {
   }
   cl_ulong total = 0;
 
+  // Take the time now in system_clock domain as a guess for the end of the run.
+  // TODO: synchronize clocks better
+  int64_t tsOffset =
+      std::chrono::system_clock().now().time_since_epoch().count();
+
+  if (!kernelLaunches_.empty()) {
+    auto &event = kernelLaunches_.back().event_;
+    cl_ulong time_end;
+    clGetEventProfilingInfo(event, CL_PROFILING_COMMAND_END, sizeof(time_end),
+                            &time_end, NULL);
+    tsOffset -= (time_end / 1000);
```

Comment:
Consider adding some comments here to explain what this `tsOffset` is for. I'm not sure that it's obvious that the host and device could be on different clocks or that individual device readings are not meaningful in this case.